packages/react-select/src/components/Select/useSelectStyles.ts

```diff
@@ -12,36 +13,197 @@ export const selectClassNames: SlotClassNames<SelectSlots> = {
-  wrapper: {
-    // TODO: add styles
+// TODO(sharing) use theme values once available
+const horizontalSpacing = {
+  xxs: '2px',
+  xs: '4px',
+  sNudge: '6px',
+  s: '8px',
+  mNudge: '10px',
+  m: '12px',
+};
+const contentSizes = {
+  // TODO: borrowed this from Input, should be in the theme somewhere?
+  body1: {
+    fontSize: tokens.fontSizeBase300,
+    lineHeight: tokens.lineHeightBase300,
+  },
+  caption1: {
+    fontSize: tokens.fontSizeBase200,
+    lineHeight: tokens.lineHeightBase200,
   },
-  select: {
-    // TODO: add styles
+  400: {
+    fontSize: tokens.fontSizeBase400,
+    lineHeight: tokens.lineHeightBase400,
   },
-  selectDisabled: {
-    // TODO: add styles
+};
+// TODO: borrowed this from Input (Select has the same size in design comps)
+// Should be in the theme somewhere?
+const fieldHeights = {
+  small: '24px',
+  medium: '32px',
+  large: '40px',
+};
+
+// TODO: borrowed from Input, also seems like animation should be included in the theme:
+const motionDurations = {
+  ultraFast: '0.05s',
+  normal: '0.2s',
+};
+const motionCurves = {
+  accelerateMid: 'cubic-bezier(0.7,0,1,0.5)',
+  decelerateMid: 'cubic-bezier(0.1,0.9,0.2,1)',
+};
+
+const useRootStyles = makeStyles({
+  base: {
+    alignItems: 'center',
+    boxSizing: 'border-box',
+    display: 'flex',
+    flexWrap: 'nowrap',
+    fontFamily: tokens.fontFamilyBase,
+    position: 'relative',
+
+    '*, *:before, *:after': {
+      boxSizing: 'border-box',
+    },
+
+    '&:after': {
+      backgroundImage: `linear-gradient(
+        0deg,
+        ${tokens.colorCompoundBrandStroke} 0%,
+        ${tokens.colorCompoundBrandStroke} 50%,
+        transparent 50%,
+        transparent 100%
+      )`,
+      ...shorthands.borderRadius(0, 0, tokens.borderRadiusMedium, tokens.borderRadiusMedium),
+      content: '""',
+      height: tokens.borderRadiusMedium,
+      position: 'absolute',
+      bottom: '0',
+      left: '0',
+      right: '0',
+      transform: 'scaleX(0)',
+      transitionProperty: 'transform',
+      transitionDuration: motionDurations.ultraFast,
+      transitionDelay: motionCurves.accelerateMid,
+    },
+
+    '&:focus-within::after': {
+      transform: 'scaleX(1)',
+      transitionProperty: 'transform',
+      transitionDuration: motionDurations.normal,
+      transitionDelay: motionCurves.decelerateMid,
+    },
+  },
+});
+
+const useSelectStyles = makeStyles({
+  base: {
+    appearance: 'none',
+    ...shorthands.border('1px', 'solid', 'transparent'),
+    ...shorthands.borderRadius(tokens.borderRadiusMedium),
+    boxShadow: 'none',
+    color: tokens.colorNeutralForeground1,
+    flexGrow: 1,
+    fontFamily: tokens.fontFamilyBase,
+
+    ':focus-visible': {
+      outlineWidth: '2px',
+      outlineStyle: 'solid',
+      outlineColor: 'transparent',
+    },
+  },
+  disabled: {
+    backgroundColor: tokens.colorTransparentBackground,
+    color: tokens.colorNeutralForegroundDisabled,
+    cursor: 'not-allowed',
```

Comment:
Should we do not-allowed on other clickable controls when they are disabled?  I didn't do that on tabs, but maybe I should.